From 2e2e841c8cad2d46433951468bead06e6a2dfbf9 Mon Sep 17 00:00:00 2001
From: kszaq <kszaquitto@gmail.com>
Date: Wed, 20 Mar 2019 14:09:41 +0100
Subject: [PATCH] GL: Don't use SM_TEXTURE_LIM for GLSL < 1.5

SM_TEXTURE_LIM is not available for GLSL < 1.5, we should use SM_TEXTURE instead.

Fixes https://github.com/xbmc/xbmc/issues/15723
---
 .../VideoPlayer/VideoRenderers/OverlayRendererGL.cpp     | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGL.cpp b/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGL.cpp
index c0223e66d68b..a75db3ca054c 100644
--- a/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGL.cpp
+++ b/xbmc/cores/VideoPlayer/VideoRenderers/OverlayRendererGL.cpp
@@ -514,7 +514,14 @@ void COverlayTextureGL::Render(SRenderState& state)
 
 #if defined(HAS_GL)
   CRenderSystemGL* renderSystem = dynamic_cast<CRenderSystemGL*>(CServiceBroker::GetRenderSystem());
-  renderSystem->EnableShader(SM_TEXTURE_LIM);
+
+  int glslMajor, glslMinor;
+  renderSystem->GetGLSLVersion(glslMajor, glslMinor);
+  if (glslMajor >= 2 || (glslMajor == 1 && glslMinor >= 50))
+    renderSystem->EnableShader(SM_TEXTURE_LIM);
+  else
+    renderSystem->EnableShader(SM_TEXTURE);
+
   GLint posLoc = renderSystem->ShaderGetPos();
   GLint tex0Loc = renderSystem->ShaderGetCoord0();
   GLint uniColLoc = renderSystem->ShaderGetUniCol();
